### ISSUE TITLE
Svg Bug On Blazor With Rectangles

### ISSUE THREAD

**Issue by musictopia2** [dotnet/aspnetcore#43858]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

When I have a drawing of a rectangle that when I have a function to select/unselect, then it gives a runtime error.
I have the sample also do circles and triangles.  In those cases, no problem.
I even have 2 pages where one is selected to begin with and another one is unselected to begin with.
In those cases, its correct to begin with.
The issue is only with rectangles.

### Expected Behavior

What I expect is to have the change done to the image that contains the rectangle instead of runtime error.

### Steps To Reproduce

Go to this github repository that duplicates the problem everytime.
https://github.com/musictopia2/SvgRectangleBug 

Click the 2 rectangle pages.
Then click on the image.
Everytime, the runtime error happens.

However, if you click the circles or the triangles, then it does what is expected.

### Exceptions (if any)

Error: Cannot remove attribute from non-element child

### .NET Version

6

**Comment by mkArtakMSFT** [comment]
Thanks for contacting us.
The error usually indicates that the DOM has been altered outside of the Blazor context.
Do you happen to do something like that?

If not, we'll try to look into this at a later time as this doesn't seem to be a common issue.

**Comment by SteveSandersonMS** [comment]
@musictopia2 Could you provide a minimal repro that doesn't depend on external packages? It looks like the code that throws is in `BasicBlazorLibrary` or `SvgHelper.Blazor`, to which we don't have sources. If you could please show the minimal code that triggers the issue, without any external dependencies, we'll be able to take a look.

**Comment by musictopia2** [comment]
I did update the repository here
https://github.com/musictopia2/SvgRectangleBug
which now uses no external dependencies and the problem still occured.

**Comment by SteveSandersonMS** [comment]
I checked the code there and unfortunately it uses some unsupported APIs and problematic patterns. The `RenderTreeBuilder` APIs aren't technically supported for public use, though if you can produce a clear minimal repro I'm still totally willing to investigate. More problematic is [this `k++`-type code](https://github.com/musictopia2/SvgRectangleBug/blob/master/SvgHelpers/SvgRenderClass.cs#L55) - see https://gist.github.com/SteveSandersonMS/ec232992c2446ab9a0059dd0fbc5d0c3.

If you are able to reduce this to a truly minimal example I'll certainly investigate. That means producing the smallest number of lines of code that triggers the problem. This definitely doesn't involve a whole general-purpose SVG rendering system, as it could be reduced to just the framework API calls that it would perform in this specific case. Otherwise there's a very strong chance that the product engineering team would just be debugging this custom SVG rendering system (which may well be where the bug is) rather than spending our resources on improving the product. Hope that makes sense and seems reasonable!

**Comment by musictopia2** [comment]
I was able to reduce to a minimal example without any custom rendering.  Everything was hard coded in the BuildRenderTree code and the same thing happened.

**Comment by SteveSandersonMS** [comment]
Thanks @musictopia2. That simplified repro helps immensely. I was able to find the problem immediately.

There is a bug in the framework **and** there is a bug in your code. Fixing either of the bugs is sufficient to make it work.

For the framework bug, I've filed https://github.com/dotnet/aspnetcore/issues/43949

The bug in your code is that your sequence numbers are wrong and don't refer to the logical sequence in the source file.  These three lines all have the same sequence number `0`:

 1. https://github.com/musictopia2/SvgRectangleBug/blob/master/Components/SimpleGraphics.cs#L37
 2. https://github.com/musictopia2/SvgRectangleBug/blob/master/Components/SimpleGraphics.cs#L52
 3. https://github.com/musictopia2/SvgRectangleBug/blob/master/Components/SimpleGraphics.cs#L67

But they all need to have different sequence numbers if you want diffing to understand they are not the same line of code. If you change them to `1`, `2`, and `3`, then the error will not occur.

I'll close this as the framework issue is now tracked in #43949
